### PR TITLE
Globalnet architecture details

### PR DESCRIPTION
**Pull request by sridhargaddam**
Fixes Issue: https://github.com/submariner-io/submariner-website/issues/20

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: src/content/architecture/globalnet/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+This is achieved by enhancing [lighthouse](https://github.com/submariner-io/lighthouse) with support for GlobalNet. The Lighthouse controller adds the service's GlobalIp to the `MultiClusterService` object that is distributed to all clusters. The [lighthouse plugin](https://github.com/submariner-io/lighthouse/tree/master/plugin/lighthouse) then uses the service's GlobalIp when replying to DNS queries for the service.
+
+## Installation
+
+TBD.
+
+## Testing
+
+Refer [testing guide](testing.md) for instructions on how to test, same steps apply.
```
Comment (mangelajo) [inline src/content/architecture/globalnet/_index.en.md:66]: This doesn't apply here.

File: src/content/architecture/globalnet/_index.en.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+### Service Discovery - Lighthouse
+
+Connectivity is only part of the solution as pods still need to know the IPs of services on remote clusters.
+
+This is achieved by enhancing [lighthouse](https://github.com/submariner-io/lighthouse) with support for GlobalNet. The Lighthouse controller adds the service's GlobalIp to the `MultiClusterService` object that is distributed to all clusters. The [lighthouse plugin](https://github.com/submariner-io/lighthouse/tree/master/plugin/lighthouse) then uses the service's GlobalIp when replying to DNS queries for the service.
+
+## Installation
+
+TBD.
```
Comment (mangelajo) [inline src/content/architecture/globalnet/_index.en.md:62]: May be we can add a link to the existing quickstart, or may be better, remove the Section "Installation" altogether.

File: src/content/architecture/globalnet/_index.en.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## Testing
+
+Refer [testing guide](testing.md) for instructions on how to test, same steps apply.
+
+## Known Issues
```
Comment (mangelajo) [inline src/content/architecture/globalnet/_index.en.md:68]: Also remove the next sections, specially the TODO if it's already captured in issues it should be fine I think.

File: src/content/architecture/globalnet/_index.en.md (hunk 1 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:66]: Actually this is taken as is from the globalnet readme we had in Submariner repo. I can remove it, no issues. Thanks for reviewing @mangelajo

File: src/content/architecture/globalnet/_index.en.md (hunk 3 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:64]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 2 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:62]: Currently, we only have a quickstart for OpenShift + Globalnet. Once we have the quickstart for KIND, we can link it here.

Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md]: Done, removed the section for now.

File: src/content/architecture/globalnet/_index.en.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+title = "Globalnet Controller"
+date = 2020-02-19T21:08:37+01:00
+weight = 5
++++
+## Introduction
+
+Submariner is a tool built to connect overlay networks of different
+Kubernetes clusters. These clusters can be on different public clouds
+or on-premise. An important use case for Submariner is to connect disparate
+independent clusters into a single cohesive multi-cluster.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:11]:
```suggestion
independent clusters into a single cohesive multi-cluster fleet. 
```

File: src/content/architecture/globalnet/_index.en.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+Submariner is a tool built to connect overlay networks of different
+Kubernetes clusters. These clusters can be on different public clouds
+or on-premise. An important use case for Submariner is to connect disparate
+independent clusters into a single cohesive multi-cluster. 
+
+However, one key limitation of submariner's current design is that it doesn't
+support overlapping CIDRs (ServiceCIDR and ClusterCIDR) across these clusters.
+Each cluster must use distinct CIDRs that don't conflict or overlap with any
+other cluster that is going to be part of cluster fleet.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:16]: The wording here sounds more like a proposal to handle overlapping CIDRs but Globalnet has already landed. Perhaps change accordingly to:

 _However, by default, a limitation of submariner is that it doesn't handle
overlapping CIDRs (ServiceCIDR and ClusterCIDR) across these clusters.
Each cluster must use distinct CIDRs that don't conflict or overlap with any
other cluster that is going to be part of cluster fleet._

File: src/content/architecture/globalnet/_index.en.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+![Figure 1 - Problem with overlapping CIDRs](/images/globalnet/overlappingcidr-problem.png)
+
+This is largely problematic because most actual deployments use the default CIDRs for a cluster so every cluster ends up using the same CIDRs. Changing CIDRs on existing clusters is a very disruptive process and requires a cluster restart. So submariner needs a way to allow clusters with overlapping CIDRs to connect together.
+
+## Architecture
+
+To support overlapping CIDRs in clusters connected through submariner, we're introducing a new component called Global Private Network, GlobalNet (`globalnet`). This GlobalNet is a virtual network specifically to support submariner's multi-cluster solution with a Global CIDR. Each cluster is given a subnet from this Global Private Network, configured as new cluster parameter `GlobalCIDR` (e.g. 169.254.0.0/16) which is configurable at time of deployment
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:24]:
```suggestion
To support overlapping CIDRs in clusters connected through submariner, submariner has a component called Global Private Network, GlobalNet (`globalnet`). This GlobalNet is a virtual network specifically to support submariner's multi-cluster solution with a Global CIDR. Each cluster is given a subnet from this Global Private Network, configured as new cluster parameter `GlobalCIDR` (e.g. 169.254.0.0/16) which is configurable at time of deployment
```

File: src/content/architecture/globalnet/_index.en.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+Once configured, each service and pod that requires cross-cluster access is allocated an IP, `GlobalIp` from this `GlobalCIDR` and it is annotated on the Pod/Service object. This `GlobalIp` is used for all cross-cluster communication to and from to this Pod/Service. Routing and IPTable/OVS/OVN rules are configured to use this IP for ingress and egress. All address translations occur on the Gateway node.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:26]:
```suggestion
Once configured, each service and pod that requires cross-cluster access is allocated an IP, named `GlobalIp`, from this `GlobalCIDR` that is annotated on the Pod/Service object. This `GlobalIp` is used for all cross-cluster communication to and from to this Pod/Service. Routing and IPTable/OVS/OVN rules are configured to use this IP for ingress and egress. All address translations occur on the Gateway node.
```

File: src/content/architecture/globalnet/_index.en.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+![Figure 1 - Proposed solution](/images/globalnet/overlappingcidr-solution.png)
+
+### submariner-globalnet
+
+Submariner GlobalNet is a component that provides cross-cluster connectivity between pods and services using their GlobalIps. Compiled as binary `submariner-globalnet`, it is the controller that is responsible for maintaining a pool of global IPs, allocating IPs from the globalIp pool to pods and services, annotating services and pods with their global Ip, and configuring the required rules on gateway node to provide cross-cluster connectivity using GlobalIps. It mainly consists of two key components: the IP Address Manager and Globalnet.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:32]:
```suggestion
Submariner GlobalNet is a component that provides cross-cluster connectivity between pods and services using their GlobalIps. Compiled as binary `submariner-globalnet`, it is responsible for maintaining a pool of global IPs, allocating IPs from the globalIp pool to pods and services, annotating services and pods with their GlobalIp, and configuring the required rules on the gateway node to provide cross-cluster connectivity using GlobalIps. It mainly consists of two key components: the IP Address Manager and Globalnet.
```

File: src/content/architecture/globalnet/_index.en.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+#### IP Address Manager (IPAM)
+
+The IP Address Manager (IPAM) component does the following:
+
+* Creates a pool of IP addresses based on `GlobalCIDR` configured on cluster.
```
Comment (tpantelis) [inline src/content/architecture/globalnet/_index.en.md:38]:
```suggestion
* Creates a pool of IP addresses based on the `GlobalCIDR` configured on cluster.
```

File: src/content/architecture/globalnet/_index.en.md (hunk 5 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:16]: Right @tpantelis. This PR basically moves the existing content from Readme to the website. Your point is valid though, I'll update the PR.

File: src/content/architecture/globalnet/_index.en.md (hunk 6 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:24]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 7 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:26]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 8 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:32]: Done

File: src/content/architecture/globalnet/_index.en.md (hunk 9 above)
Comment (sridhargaddam) [reply src/content/architecture/globalnet/_index.en.md:38]: Done

**Pull request merged**